apps/src/dropletUtils.js

```diff
@@ -157,41 +159,47 @@ standardConfig.categories = {
 };
 
 /**
- * @param codeFunctions
+ * @param codeFunctions {object} A collection of named key/value pairs
+ *   key is a block name from dropletBlocks or standardBlocks
+ *   value is an object that can be used to override block defaults
  * @param {DropletConfig} dropletConfig
  * @param {DropletConfig} otherConfig optionally used to supply a standardConfig
  *  object which is not app specific. It will be used first, then overriden
  *  by the primary dropletConfig if there is overlap between the two.
  * @param paletteOnly boolean: ignore blocks not in codeFunctions palette
- * @returns {Array}
+ * @returns {Array<DropletBlock>}
  */
-function mergeFunctionsWithConfig(codeFunctions, dropletConfig, otherConfig, paletteOnly) {
+function filteredBlocksFromConfig(codeFunctions, dropletConfig, otherConfig, paletteOnly) {
```

Comment:
I don't think that as it's currently written it should be modifying any config, should it? It's just assembling a list of blocks. Of course, if you modify the returned blocks, you will also  be modifying the version referenced by the config. Is that what you're referring to?